api/capi/src/nnstreamer-capi-single-new.c

```diff
@@ -0,0 +1,500 @@
+    /* Validate input tensor info. */
+    if (ml_tensors_info_validate (input_info, &valid) != ML_ERROR_NONE ||
+        valid == false) {
+      ml_loge ("The given param, input tensor info is invalid.");
+      return ML_ERROR_INVALID_PARAMETER;
+    }
+  }
+
+  if (output_info) {
+    /* Validate output tensor info. */
+    if (ml_tensors_info_validate (output_info, &valid) != ML_ERROR_NONE ||
+        valid == false) {
+      ml_loge ("The given param, output tensor info is invalid.");
+      return ML_ERROR_INVALID_PARAMETER;
+    }
+  }
+
+  /**
+   * 1. Determine nnfw
+   */
+  if ((status = ml_single_check_nnfw (model, &nnfw)) != ML_ERROR_NONE)
+    return status;
+
+  /**
+   * 2. Determine hw
+   * @todo Now the param hw is ignored.
+   * (Supposed CPU only) Support others later.
+   */
+  status = ml_check_nnfw_availability (nnfw, hw, &available);
+  if (status != ML_ERROR_NONE)
+    return status;
+
+  if (!available) {
+    ml_loge ("The given nnfw is not available.");
+    status = ML_ERROR_NOT_SUPPORTED;
+    return status;
+  }
+
+  /** Create ml_single object */
+  single_h = g_new0 (ml_single, 1);
+  g_assert (single_h);
+  single_h->filter = g_object_new (G_TYPE_TENSOR_FILTER_SINGLE, NULL);
+  if (single_h->filter == NULL) {
+    status = ML_ERROR_INVALID_PARAMETER;
+    goto error;
+  }
+  filter_obj = G_OBJECT (single_h->filter);
+
+  /**
+   * 3. Construct a pipeline
+   * Set the pipeline desc with nnfw.
+   */
+  switch (nnfw) {
+    case ML_NNFW_TYPE_CUSTOM_FILTER:
+      g_object_set (filter_obj, "framework", "custom",
+          "model", model, NULL);
+      break;
+    case ML_NNFW_TYPE_TENSORFLOW_LITE:
+      /* We can get the tensor meta from tf-lite model. */
+      g_object_set (filter_obj, "framework", "tensorflow-lite",
+          "model", model, NULL);
+      break;
+    case ML_NNFW_TYPE_TENSORFLOW:
+      if (in_tensors_info && out_tensors_info) {
+        ml_single_set_inout_tensors_info (filter_obj, "input",
+            in_tensors_info);
+        ml_single_set_inout_tensors_info (filter_obj, "output",
+            out_tensors_info);
+        g_object_set (filter_obj, "framework", "tensorflow",
+            "model", model, NULL);
+      } else {
+        ml_loge ("To run the pipeline with tensorflow model, \
+            input and output information should be initialized.");
+        status = ML_ERROR_INVALID_PARAMETER;
+        goto error;
+      }
+      break;
+    default:
+      /** @todo Add other fw later. */
+      ml_loge ("The given nnfw is not supported.");
+      status = ML_ERROR_NOT_SUPPORTED;
+      goto error;
+  }
+
+  /* 4. Allocate */
+  ml_tensors_info_initialize (&single_h->in_info);
+  ml_tensors_info_initialize (&single_h->out_info);
+
+  /* 5. Start the nnfw to egt inout configurations if needed */
+  klass = g_type_class_peek (G_TYPE_TENSOR_FILTER_SINGLE);
+  if (!klass) {
+    status = ML_ERROR_INVALID_PARAMETER;
+    goto error;
+  }
+  status = klass->start (single_h->filter);
```

Comment:
klass->start() does not return ml-error code. IMO it just checks bool result like line 282. 
!klass->start (single_h->filter)

Reply:
Updated to directly check the bool result.